### DISCUSSION TITLE
Multiple base URLs

### DISCUSSION THREAD

**Discussion by RealOrangeOne** [plausible/analytics#4272] [Self-Hosted Support] [answered]
My deployment consists of a public domain, obscured to prevent being hit by ad blockers, and a more sensible domain for the console. In version 2.1, this stops working, as parts of the application now check specific headers for the correct host. After poking overriding some headers in `nginx`, I eventually got something working, besides websockets.

It would be nice if the `BASE_URL` was still used for certain things, but it was possible to access the application on other hostnames (specified through another environment variable).

I know other web frameworks definitely support something like this, although I'm not familiar enough with Phoenix to implement this myself.

**Comment by ruslandoga** [comment, marked as answer]
👋 @RealOrangeOne 

The setup you described should still be possible, please see https://github.com/plausible/analytics/issues/4126

BASE_URL would be the URL of the dashboard, and the domain for /api/events can be anything since it doesn't require cookies.

**Reply by RealOrangeOne** [reply]
Aha, you're absolutely right! Digging deeper, in https://github.com/plausible/analytics/pull/1048 it became no longer necessary to have `BASE_URL` be the URL of the API - it uses the script's origin for that instead. Changing the `BASE_URL` to the dashboard URL seems to work perfectly. Thanks!
